pkg/build/generator/generator.go

```diff
@@ -432,6 +432,14 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	// Retry getting a non-conflicting name 10 times
+	if _, err := g.Client.GetBuild(ctx, build.Name); err == nil {
+		for i := 0; i < 10 && err == nil; i++ {
+			glog.V(5).Infof("A naming collision occurred with %s, retrying", build.Name)
```

Comment:
Please add the namespace to the log, otherwise we won't know what build we're talking about